### PR TITLE
Allow persistent storage for managed etcd

### PR DESCRIPTION
**Pull request by sjenning**
This allow a HostedCluster creator to specify a `PersistentVolumeClaimSpec` that managed etcd will use for persistent storage

@derekwaynecarr @csrwng

### CODE REVIEW

File: api/v1alpha1/hostedcluster_types.go [hunk 1]
```diff
@@ -379,6 +379,10 @@ type EtcdSpec struct {
 type ManagedEtcdSpec struct {
 
 	//TODO: Ultimately backup policies, etc can be defined here.
+
+	// PVCSpec provides a PersistentVolumeClaimSpec that allows the managed etcd to use persistent storage
+	// +optional
+	PVCSpec *corev1.PersistentVolumeClaimSpec `json:"pvcSpec,omitempty"`
```
Comment (derekwaynecarr) [inline api/v1alpha1/hostedcluster_types.go:385]: do we want or need the full spec?

Comment (derekwaynecarr) [reply api/v1alpha1/hostedcluster_types.go:385]: we should avoid abbreviations per API conventions.
see: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-conventions

Comment (ironcladlou) [reply api/v1alpha1/hostedcluster_types.go:385]: Attempt at minimal set of fields:
- storage class
- access modes
- size

wdyt?

Comment (ironcladlou) [reply api/v1alpha1/hostedcluster_types.go:385]: The access mode might be dependent on the requirements of the implementation we use; seems likely it'll always be `ReadWriteOnce` in practice, so maybe leave it unless there's a clear need and document that ReadWriteOnce is the fixed value?

**Pull request merged**